### PR TITLE
Updated discovery for new FHIR resources

### PR DESCRIPTION
**Pull request by dogversioning**
This adds allergyintolerance, diagnosticreport, and procedure to resources the discovery study checks for coding information.

It also lightly redoes test data generation/validation logic due to some issues with commas inside fields.

### Checklist
- [X] Consider if documentation in `docs/` needs to be updated
  - If you've changed the structure of a table, you may need to run `generate-md`
  - If you've added/removed `core` study fields that not in US Core, update our list of those in `core-study-details.md`
- [X] Consider if tests should be added
- [X] Update template repo if there are changes to study configuration in `manifest.toml`

### CODE REVIEW

File: cumulus_library/studies/discovery/reference_sql/code_detection.sql [hunk 1]
```diff
@@ -7,6 +7,90 @@
...
 
 CREATE TABLE discovery__code_sources AS
+SELECT *
+FROM (
+    VALUES (
+        'allergyintolerance',
+        'clinicalstatus',
+        '',
+        '',
+        ''
```
Comment (mikix) [inline cumulus_library/studies/discovery/reference_sql/code_detection.sql:17]: This reference file has a lot of blanks - is there no coding data in the test data for allergies or these others? Procedure.reasonCode has something. But it's a rarity. Even `encounter.hospitalization.dischargedisposition` lost data.

**Review by mikix — APPROVED** [review]

File: cumulus_library/studies/discovery/reference_sql/code_detection.sql (hunk 1 above)
Comment (dogversioning) [reply cumulus_library/studies/discovery/reference_sql/code_detection.sql:17]: This is probably due to running against my test db - my internet access has been so bad from home over the last few days on VPN that i got bizaare timeouts trying to run this against cerner. I'll try it again tomorrow when i'm onsite.

File: tests/test_discovery.py [hunk 2]
```diff
@@ -45,30 +46,30 @@ def test_discovery(tmp_path):
     db = databases.DuckDatabaseBackend(f"{tmp_path}/duck.db")
     cursor = db.cursor()
     table_rows, cols = conftest.get_sorted_table_data(cursor, "discovery__code_sources")
+    for i in range(0, len(table_rows)):
+        table_rows[i] = tuple([x if x else "" for x in table_rows[i]])
```
[the remark this answers is not in the document]
Comment (dogversioning) [reply tests/test_discovery.py:50]: i personally find this more obtuse, but i agree it's more pythonic, so i'll make this change.

File: tests/test_discovery.py [hunk 3]
```diff
@@ -45,30 +46,30 @@ def test_discovery(tmp_path):
...
-                assert tuple(ref_row) in table_rows
-        except Exception as e:
-            conftest.debug_diff_tables(cols, table_rows, ref, pos=0)
-            raise e
+        "/test_data/discovery/discovery__code_sources.csv",
+        "w",
+    ) as f:
+        writer = csv.writer(f, quoting=csv.QUOTE_MINIMAL)
+        for row in table_rows:
+            writer.writerow(row)
```
[the remark this answers is not in the document]
Comment (dogversioning) [reply tests/test_discovery.py:60]: I could've sworn i'd done this.

File: cumulus_library/studies/discovery/reference_sql/code_detection.sql (hunk 1 above)
Comment (mikix) [reply cumulus_library/studies/discovery/reference_sql/code_detection.sql:17]: I think in CONTRIBUTING.md, we have a one-liner to run against the test data - does that generate good-enough results?

File: tests/test_discovery.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikix) [reply tests/test_discovery.py]: I only generally like that construction with good formatting:
```
table_rows = [
    tuple(x or "" for x in row)
    for row in table_rows
]
```

Which I find readable enough - but if you vibe your approach for readability, I think that's a good enough reason to keep it as-is.

File: cumulus_library/studies/discovery/reference_sql/code_detection.sql (hunk 1 above)
Comment (dogversioning) [reply cumulus_library/studies/discovery/reference_sql/code_detection.sql]: updated w/ local db - still pretty blank-ful.

**Pull request merged**